backend/routes/quests.go

```diff
@@ -211,6 +220,103 @@ func GetMainUserQuests(w http.ResponseWriter, r *http.Request) {
 	routeutils.WriteDataJson(w, string(quests))
 }
 
+func GetDailyQuestStatusProgress(w http.ResponseWriter, r *http.Request) {
+	userAddress := r.URL.Query().Get("address")
+	if userAddress == "" {
+		routeutils.WriteErrorJson(w, http.StatusBadRequest, "Missing address parameter")
+		return
+	}
+
+	dayIndexStr := r.URL.Query().Get("dayIndex")
+	if dayIndexStr == "" {
+		routeutils.WriteErrorJson(w, http.StatusBadRequest, "Missing dayIndex parameter")
+		return
+	}
+
+	dayIndex, err := strconv.Atoi(dayIndexStr)
+	if err != nil {
+		routeutils.WriteErrorJson(w, http.StatusBadRequest, "Invalid dayIndex parameter")
+		return
+	}
+
+	questJsons, err := core.PostgresQueryJson[QuestProgress](
+		`SELECT quest_id AS QuestId, quest_type as QuestType
+         FROM DailyQuests
+         WHERE day_index = $1`,
+		dayIndex,
+	)
+
+	if err != nil {
+		routeutils.WriteErrorJson(w, http.StatusInternalServerError, "Failed to get daily quest status")
+		return
+	}
+
+	var quests []QuestProgress
+	err = json.Unmarshal([]byte(questJsons), &quests)
+	if err != nil {
+		routeutils.WriteErrorJson(w, http.StatusInternalServerError, "Failed to unmarshal response")
+		return
+	}
+
+	var result []QuestProgress
+	for _, quest := range quests {
+		result = append(result, QuestProgress{
+			QuestId:  quest.QuestId,
+			Progress: 60,
+			Needed:   40,
+		})
+	}
+
+	jsonResult, err := json.Marshal(result)
+	if err != nil {
+		routeutils.WriteErrorJson(w, http.StatusInternalServerError, "Failed to marshal response")
+		return
+	}
+
+	routeutils.WriteDataJson(w, string(jsonResult))
+}
+
+func GetMainQuestStatusProgress(w http.ResponseWriter, r *http.Request) {
```

Comment:
Please apply the same changes mentioned from the other function to this one aswell